### ISSUE TITLE
Breaking behavior in 2.9.0 release: terraform validate fails because of missing token

### ISSUE THREAD

**Issue by mariux** [terraform-providers/terraform-provider-github#503]
Hi there, 

As mentioned here https://github.com/terraform-providers/terraform-provider-github/pull/464#issuecomment-651254293, we are facing breaking behavior in a minor release.

We applied the workaround mentioned below and are fine for now, but I want to raise the issue anyway as we faced this in our automated testing of the github open-source modules we provide at @mineiros-io to make managing github configuration easy.

## Terraform & Provider Version
```
Terraform v0.12.28
+ provider.github v2.9.0
```

### Affected Resource(s)
- github provider config

### Terraform Configuration Files
N/A

### Debug Output
With `2.8.1` `terraform validate` in our module code validates just fine. With `2.9.0` we get the following breaking error now:
```
$ terraform-0.12.28 validate

Error: Missing required argument

The argument "token" is required, but was not set.
```

### Expected Behavior
- `terraform validate` just run fine without carping about a missing `token`.

### Actual Behavior
- `terraform validate` fails with `The argument "token" is required, but was not set.`
- **current workaround**: Setting `GITHUB_TOKEN` environment to a random string fixes the behavior.

### Steps to Reproduce
- run `terraform validate` without specifying a token configuration with 2.8.1 and 2.9.0

### References
The issue was raised before in the introducing PR introducing the breaking change:
- #464
- #502

**Comment by glenwinters** [comment]
I'd say this is related to https://github.com/terraform-providers/terraform-provider-github/issues/502 since the token was required unless `anonymous` was true according to the docs.

The fix for now is to pin the provider to the previous release, 2.8.1.

**Comment by mariux** [comment]
> @mariux what could you do with anonymous GitHub provider before?

We provide a module and run automated tests for static analysis and unit testing. The static analysis runs tools like `tflint` and `terraform validate`. `terraform validate` started failing with 2.9.0 if no token is provided. As this is a module we do not configure the provider at all but just have a version constraint defined:

```
terraform {
  required_version = "~> 0.12.9"

  required_providers {
    github = "~> 2.4"
  }
}
```

(crossposted from #464) to clarify the details.

**Comment by mariux** [comment]
> I'd say this is related to #502 since the token was required unless `anonymous` was true according to the docs.
> The fix for now is to pin the provider to the previous release, 2.8.1.

Added #502 to the list of references, thanks. For this specific issue, we worked around by setting the TOKEN via environment variables also for the validation step.

**Comment by glenwinters** [comment]
Here's how I'm using the `anonymous` attribute:

```terraform
provider "github" {
  anonymous  = true
  individual = true
}

data "github_ip_ranges" "default" {}
```

I'm using the ip ranges returned by that data block to give GitHub hooks access to my API.

I'm not setting the token using an environment variable.

**Comment by jcudit** [comment]
Changes to unblock this are now merged into the default branch.  With the v3.0.0 release likely landing mid next week.

**Comment by marco-m-pix4d** [comment]
Hello, the v3.0.0 release is out, does it mean that this bug is fixed ?

**Comment by jcudit** [comment]
Yes, this should be resolved in the latest release. Please re-open if there are further issues.

**Comment by lyhoanglong** [comment]
Thi bug is not fix. I still see the bug
```
Error: Missing required argument

The argument "token" is required, but was not set.
```

```
provider "gitlab" {
  token = "something"
}
```

```
terraform {
  required_providers {
    gitlab = {
      source = "gitlabhq/gitlab"
    }
  }
  required_version = ">= 0.13"
}

```
